vmm/src/default_syscalls/x86_64.rs

```diff
@@ -279,55 +279,7 @@ pub fn default_context() -> Result<SeccompFilterContext, Error> {
                 ),
             ),
             allow_syscall(libc::SYS_munmap),
-            (
-                libc::SYS_open,
-                (
-                    0,
-                    vec![
-                        SeccompRule::new(vec![], SeccompAction::Allow),
-                        SeccompRule::new(
-                            vec![SeccompCondition::new(1, SeccompCmpOp::Eq, O_RDWR)?],
-                            SeccompAction::Allow,
-                        ),
-                        SeccompRule::new(
-                            vec![SeccompCondition::new(
-                                1,
-                                SeccompCmpOp::Eq,
-                                O_RDWR | O_CLOEXEC,
-                            )?],
-                            SeccompAction::Allow,
-                        ),
-                        SeccompRule::new(
-                            vec![SeccompCondition::new(
-                                1,
-                                SeccompCmpOp::Eq,
-                                O_RDWR | O_NONBLOCK | O_CLOEXEC,
-                            )?],
-                            SeccompAction::Allow,
-                        ),
-                        SeccompRule::new(
-                            vec![SeccompCondition::new(1, SeccompCmpOp::Eq, O_RDONLY)?],
-                            SeccompAction::Allow,
-                        ),
-                        SeccompRule::new(
-                            vec![SeccompCondition::new(
-                                1,
-                                SeccompCmpOp::Eq,
-                                O_RDONLY | O_CLOEXEC,
-                            )?],
-                            SeccompAction::Allow,
-                        ),
-                        SeccompRule::new(
-                            vec![SeccompCondition::new(
-                                1,
-                                SeccompCmpOp::Eq,
-                                O_RDONLY | O_NONBLOCK | O_CLOEXEC,
-                            )?],
-                            SeccompAction::Allow,
-                        ),
-                    ],
-                ),
-            ),
+            allow_syscall(libc::SYS_open),
```

Comment:
Are we sure we want to relax the allowed parameters for the SYS_open syscall? 
I am trying to understand why keeping the `SeccompRules` for tightening the allowed parameters and remove the one that makes them useless (i.e `SeccompRule::new(vec![], SeccompAction::Allow)`) was not the chosen approach?

Reply:
Because `open` gets called with so many combinations of flags `( O_RDONLY or O_RDWR ) [ | O_CLOEXEC ] [ | O_NONBLOCK ]` (possibly more that we haven't discovered yet) that there doesn't seem to be much to gain from overcomplicating the filter.